pkg/registry/generic/etcd/etcd.go

```diff
@@ -152,23 +152,24 @@ func (e *Etcd) ListPredicate(ctx api.Context, m generic.Matcher) (runtime.Object
 	trace := util.NewTrace("List " + reflect.TypeOf(list).String())
 	defer trace.LogIfLong(600 * time.Millisecond)
 	if name, ok := m.MatchesSingle(); ok {
-		trace.Step("About to read single object")
-		key, err := e.KeyFunc(ctx, name)
-		if err != nil {
-			return nil, err
-		}
-		err = e.Storage.GetToList(key, list)
-		trace.Step("Object extracted")
-		if err != nil {
-			return nil, err
-		}
-	} else {
-		trace.Step("About to list directory")
-		err := e.Storage.List(e.KeyRootFunc(ctx), list)
-		trace.Step("List extracted")
-		if err != nil {
-			return nil, err
+		if key, err := e.KeyFunc(ctx, name); err == nil {
```

Comment:
Shouldn't - you have to say what namespace you want to watch in.  The problem is the predictate (metadata.name) does not include the implicit predicate coming off the url (the namespace value).  KeyFunc requires name && namespace > 0.  The only thing this prevents is us rejecting metadata.name= at this level (but it should be higher anyway)